t/lib/overload_join.t

```diff
@@ -0,0 +1,162 @@
+use warnings;
+use strict;
+
+BEGIN {
+    chdir 't' if -d 't';
+    @INC = '../lib';
+    require './test.pl';
+}
+
+# This class:
+# - adds * around non-whitespace "words" when stringified
+# - assimilates strings when concatenated (i.e. always return an object)
+
+package BoldStars;
+use Data::Dumper;
+
+use overload
+  '""' => sub {
+    return ${ $_[0] } =~ s{(\S+)}{*$1*}gr;
+  },
+  '.' => sub {
+    my ( $l, $r, $s ) = @_;
```

Comment:
Tiniest of thoughts: it's a bit clearer what `$s` is doing if you name it `$swapped` or somesuch